### DISCUSSION TITLE
An error during totp-cli generate

### DISCUSSION THREAD

**Discussion by masoncole1994** [yitsushi/totp-cli#83] [Q&A] [answered]
Hi @yitsushi,

First of all, thank you for your product.

I'm seeking some help with the following situation:

```shell
totp-cli generate foo_company foo_service
Password: ***
Error: OTP error: illegal base32 data at input byte 1
```

I've added a profile like this:
```shell
totp-cli add-token
Namespace: foo_company
Account: foo_service
Token: 123789
Password: ***
```

I entered a 6-digit code from the Google Authenticator app into the `Token` field, but I suspect that it's incorrect 😄. The documentation was unclear, and I couldn't find any other relevant information in my Google account settings.

Thank you.

**Comment by yitsushi** [comment, marked as answer]
The Token is the one the company provided when you activated 2FA. Usually you can see a QR code and scan it and around the QR there are a "can't scan, show me text" or similar, that's the Token. The generated 6 numbers are one time passwords. That's the general glossary for Time-based One Time Password (TOTP).

Does it help?

As for you you can get the Token, if you search for "[appname] export secret tokens" for example "LastPass export secret tokens", most of the known applications has some ways to export that content. If not, the only way (sadly) is to generate a new one and invalidate the old one. When I first (years ago) realized I can't get that Token easily, I regenerated all my TOTP tokens, added to my authenticator and to my totp-cli, while saved all the tokens in an encrypted backup file, so I can swap totp apps easily. Now they are stored in BitWarden where I can see the token anytime, so I don't need that extra tile, but I still keep it as an encrypted file in case something happens with BitWarden.

**Reply by masoncole1994** [reply]
Thank you for the explanation. I found the token and now able to generate otp codes 🎉
